aplib.net-demo/Assets/Scripts/WFC/GameObjectPlacer.cs

```diff
@@ -159,11 +159,17 @@ public void PlaceTile(int x, int z, Tile tile, Transform parent)
         {
             GameObject prefab = tile switch
             {
+                StartCorner => _roomObjects.StartCorner,
                 Corner => _roomObjects.Corner,
+                StartCrossing => _roomObjects.StartCrossing,
                 Crossing => _roomObjects.Crossing,
+                StartDeadEnd => _roomObjects.StartDeadEnd,
                 DeadEnd => _roomObjects.DeadEnd,
+                StartRoom => _roomObjects.StartRoom,
                 Room => _roomObjects.Room,
+                StartStraight => _roomObjects.StartStraight,
                 Straight => _roomObjects.Straight,
+                StartTSection => _roomObjects.StartTSection,
                 TSection => _roomObjects.TSection,
                 _ => _roomObjects.Empty
             };
```

Comment:
You're not going to like my comment :(
The current way of implementing is not good. It uses unnecessary classes, creating a complex hierarchy. Everytime we add another tile, we need to also create a class for this in the code. Creating a separate model is fine, but creating a class is definitely unnecessary.

A better way to tackle this is to add a `IsStart` boolean property to `Tile`and add some simple logic to the `switch`-expression.
```suggestion
            GameObject prefab = tile switch
            {
                Corner c when c.IsStart => _roomObjects.StartCorner,
                Corner => _roomObjects.Corner,
                Crossing c when c.IsStart => _roomObjects.StartCrossing,
                Crossing => _roomObjects.Crossing,
                DeadEnd d when d.IsStart => _roomObjects.StartDeadEnd,
                DeadEnd => _roomObjects.DeadEnd,
                Room r when r.IsStart => _roomObjects.StartRoom,
                Room => _roomObjects.Room,
                Straight s when s.IsStart => _roomObjects.StartStraight,
                Straight => _roomObjects.Straight,
                TSection t when t.IsStart => _roomObjects.StartTSection,
                TSection => _roomObjects.TSection,
                _ => _roomObjects.Empty
            };
```


By implementing it this way, the extra classes like `StartCorner` or `StartTSection` can be removed.

Reply:
Having separate `GameObject` properties in `RoomBojects.cs` is fine, so you don't have to remove those.

Reply:
The `SwitchTile` method would have to be updated to only set the `IsStart` boolean to `true`.

Reply:
Just implemented this! Thank you for the clear instructions, this was smooth changing! Also, this is indeed better than having so many classes :D